js/client.js

```diff
@@ -1636,19 +1650,37 @@
 			document.title = room.title ? room.title + " - Showdown!" : "Showdown!";
 		},
 		updateAutojoin: function () {
-			if (Config.server.id !== 'showdown') return;
+			if (!Config.server.registered) return;
+			var curAutojoin = (Tools.prefs('autojoin') || '');
+			if (typeof curAutojoin === 'string' && Config.server.id !== 'showdown') {
+				// Switch to the autojoin object to handle multiple servers
+				curAutojoin = {showdown: curAutojoin};
+			}
 			var autojoins = [];
 			var autojoinCount = 0;
 			var rooms = this.roomList.concat(this.sideRoomList);
 			for (var i = 0; i < rooms.length; i++) {
 				var room = rooms[i];
 				if (room.type !== 'chat') continue;
+				if (room.id === 'staff' || room.id === 'upperstaff' || (Config.server.id !== 'showdown' && room.id === 'lobby')) continue;
 				autojoins.push(room.id.indexOf('-') >= 0 ? room.id : (room.title || room.id));
-				if (room.id === 'staff' || room.id === 'upperstaff') continue;
 				autojoinCount++;
 				if (autojoinCount >= 10) break;
 			}
-			Tools.prefs('autojoin', autojoins.join(','));
+			if (typeof curAutojoin !== 'string') {
+				if (curAutojoin[Config.server.id] === autojoins.join(',')) return;
+				if (!autojoins.length) {
+					delete curAutojoin[Config.server.id];
+					// If the only key left is 'showdown', revert to the string method for storing autojoin.
+					if (Object.keys(curAutojoin).length === 1 && Object.keys(curAutojoin)[0] === 'showdown') curAutojoin = curAutojoin.showdown;
```

Comment:
Is `Object.keys` supported by ES3? You should probably keep track of that separately.